content/knowledge/flare_architecure.md

```diff
@@ -0,0 +1,192 @@
+A node may be a member of arbitrarily many subnets and must be part of the primary network.
+The primary network contains three build-in blockchains: P-Chain (Platform Chain), C-Chain (Contract Chain), and X-Chain (Exchange Chain).
+
+In total, the Avalanche ecosystem (as of the end of December 2021) operates [13 subnets and 10 blockchains](https://explorer.avax.network/subnets?tab=validators) (3 main plus 7 minor ones: Dtehereum, Btehereum, C and X-chain PoW, and others).
+
+All the details regarding the platform operation, tutorials, and tools are described in the [official Avalanche website](https://docs.avax.network/build/tutorials/platform/create-a-subnet/.)
+
+#### P-Chain
+
+The Platform Chain is the metadata blockchain on Avalanche that coordinates validators, keeps track of active subnets, and enables the creation of new subnets.
+The P-Chain uses a [UTXO](https://www.investopedia.com/terms/u/utxo.asp) (Unspent Transaction Output) model and implements the Snowman consensus protocol.
+Adding or removing a validator from the validator set at some given time by the P-Chain needs absolute ordering, so parallel lines of computation (DAG approach) are not allowed.
+
+#### C-Chain
+
+With the C-Chain API, clients can create smart contracts.
+It is an example of the Ethereum Virtual Machine that the Avalanche Chain powers.
+The C-Chain also serves as the default smart contract blockchain of the Avalanche Chain.
+C-Chain is also very compatible with the Solidity smart contract and Ethereum tooling.
+Therefore, it makes it easy for Ethereum developers to port applications into the Avalanche chain easily.
+C-Chain uses an account-based Snowman Consensus Protocol since an application requires a total ordering of transactions and runs consensus on blocks, not transactions.
+That is, if it needs to be able to compare any two transactions and determine which came first, it can't use a DAG because the ordering of some transactions is not defined.
+
+#### X-Chain
+
+The X-Chain functions as a decentralized network for creating and trading digital assets and uses the Avalanche consensus protocol.
+The X-Chain uses a DAG with the full Avalanche consensus protocol, so it is UTXO (Unspent Transaction Output) based.
+UTXO-based systems are inherently more scalable than account-based systems.
+They are naturally parallelizable, whereas account-based systems must lock around the account to process transactions.
+It is a real-world representation of assets such as bonds and equity with specific rules that govern their actions.
+Whenever you initiate a transaction on the Avalanche blockchain, you get to pay a fee in AVAX.
+The Exchange Chain is an Avalanche Virtual Machine (AVM) example.
+Clients can create and trade assets on the X-Chain and other examples of the AVM with the help of the X-Chain API.
+
+## Flare's FBA
+
+_TODO_ add more details to each of the items
```

Comment:
Our rule at Optakt is that any TODO should come with a GitHub issue referenced with the TODO, while FIXMEs are for things that will not be done in an issue, but then they should be solved before the PR is ever merged.

Reply:
I've added an issue to the repo